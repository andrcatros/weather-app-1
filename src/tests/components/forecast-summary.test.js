import React from 'react'
import { render } from '@testing-library/react'
import ForecastSummary from '../../components/forecastsummary'

describe('Forecast Summary', () => {
    it('renders correctly', () => {
        const { asFragment } = render(
            <ForecastSummary
                date={999}
                temperature={68494}
                description='testDesc'
                icon='testIcon'
            />
        )

        expect(asFragment()).toMatchSnapshot();
    })

    it('renders a number of props into a weather summary', () => {
        const { getByText } = render(
            <ForecastSummary
                date={281020}
                temperature={9}
                description='Clear'
                icon='https://icon.com'
            />
        )

        expect(getByText('281020')).toBeTruthy();
        expect(getByText('281020')).toHaveClass('date');
        expect(getByText('9')).toHaveClass('max-temperature');
        expect(getByText('Clear')).toHaveClass('description');
        expect(getByText('https://icon.com')).toHaveClass('icon');
    })
})