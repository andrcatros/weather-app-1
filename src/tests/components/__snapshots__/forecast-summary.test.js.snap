// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Summary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    <span
      class="date"
      data-testid="date-id"
    >
      999
    </span>
    <span
      class="icon"
      data-testid="icon-id"
    >
      testIcon
    </span>
    <span
      class="max-temperature"
      data-testid="temperature-id"
    >
      68494
    </span>
    <span
      class="description"
      data-testid="description-id"
    >
      testDesc
    </span>
  </div>
</DocumentFragment>
`;
